jukepoot/puppet_parser.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/python
+"""A parser of /var/log/puppet.log.
+
+
+"""
+
+
+import argparse 
+from datetime import datetime
+import dateutil.parser
+import re
+import collections
+import os
+
+
+def main():
+	parser = argparse.ArgumentParser()
+	parser.add_argument(
+		help="parse the given filename", type=str, 
+		dest="filename", action="store")
+	parser.add_argument(
+		"-y","--year", help="parse file by setting start year", 
+		type=int, dest="year", action="store")
+	args = parser.parse_args()
+	if args.year:
+		print("year to use: {0:d}".format(args.year))
+		year = args.year
+
+	else: #build year for date parsing
+		year = datetime.utcfromtimestamp(
+			os.stat(args.filename).st_birthtime).year
+		# print("creation of file: {0:d}".format(year))
+
+	puppet_parser(args.filename,year)
+
+
+def puppet_parser(filepath, year):
+	"""Accepts the puppet file path and parses it out by calling other functions.
+
+	"""
+
+	# pattern to match from log file and group each finding for referencing
+	# pattern looks for a time foormat (x:x:x)any text after which is
+	# followed by the process and so on 
+	pattern = re.compile(
+		r'(^.+:..:..)\s(\S+)\s(\S+)(\[\d+\]):\s(.+$)')
+	#group 1 = date
+	#group 2 = hostname
+	#group 3 = proccess
+	#group 4 = PID
+	#group 5 = message
+	
+	# add to list for counter purposes 
+	message_list = []
+	proccess_list =[]
+	hostname_list= []
+	processID_list = []
```

Comment:
what I don't understand here is that you are adding these as lists here, and then later on reducing this to collections.Counter... instead of just using collections.Counter to begin with... would save you a lot of trouble and having this saved twice